lib/import/brca/providers/salisbury/salisbury_handler.rb

```diff
@@ -13,62 +11,262 @@ def process_fields(record)
+            fields.each do |field|
+              result = raw_record[field]&.scan(BRCA_REGEX)&.flatten&.uniq
+              if result.present?
+                status_genes = result
+                break
+              end
+            end
+            status_genes
           end
 
-          def process_molecular_testing(genotype, record)
-            mol_testing_type = record.raw_fields['moleculartestingtype']&.downcase
-            genotype.add_molecular_testing_type_strict(TEST_TYPE_MAPPING[mol_testing_type])
-            scope = TEST_SCOPE_MAPPING[mol_testing_type].presence || :no_genetictestscope
+          def assign_status_var(raw_record)
+            @status = raw_record['status']&.downcase
+          end
+
+          # Use priority if more than one status is present for same gene for a given record
+          # rubocop:disable Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/PerceivedComplexity
+          def process_multi_status_genes(status_genes, status_found, genotype, genotypes, raw_record)
+            if status_found.intersect?(POSITIVE_STATUS) || status_found.any? { |e| e.match(/^variant/) }
+              process_status_genes(status_genes, 2, genotype, genotypes, raw_record) if extract_teststatus_record == 2
+            elsif status_found.intersect?(ABNORMAL_STATUS)
+              process_status_genes(status_genes, 10, genotype, genotypes, raw_record) if extract_teststatus_record == 10
+            elsif status_found.intersect?(NEGATIVE_STATUS)
+              process_status_genes(status_genes, 1, genotype, genotypes, raw_record) if extract_teststatus_record == 1
+            elsif status_found.match(FAILED_TEST)
+              process_status_genes(@all_genes, 9, genotype, genotypes, raw_record) if extract_teststatus_record == 9
+            elsif status_found.intersect?(UNKNOWN_STATUS)
+              process_status_genes(status_genes, 4, genotype, genotypes, raw_record) if extract_teststatus_record == 4
+            end
+          end
+          # rubocop:enable Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/PerceivedComplexity
+
+          def process_status_genes(genes, status, genotype, genotypes, record)
+            return unless genes&.all? { |gene| @all_genes.include?(gene) }
+
+            genes.each do |gene|
+              @all_genes -= [gene]
+              genotype_new = genotype.dup
+              genotype_new.add_gene(gene)
+              genotype_new.add_status(status)
+              if [2, 10].include? status
+                handle_variant_record(genotype_new, record, genotypes)
+              else
+                genotypes << genotype_new
+              end
+            end
+            genotypes
+          end
+
+          def handle_variant_record(genotype_new, record, genotypes)
+            assign_variantpathclass_record(genotype_new)
+            variant = record['genotype']
+            if variant.present?
+              if variant.scan(CDNA_REGEX).size > 1 ||
```

Comment:
does this need to be (variant.scan(CDNA_REGEX).size + variant.scan(EXON_VARIANT_REGEX).size) >1

Reply:
Yes good point, have now fixed it in [9f1ea0e](https://github.com/NHSDigital/data_management_system/pull/127/commits/9f1ea0e46d21ab3490e256cf066b089d360a440e)